File: Native/Test/Fabric/event.js
F = require('Fabric').createClient();

var mapNamedObjectsToNames = function (namedObjects) {
  var result = [];
  for (var i=0; i<namedObjects.length; ++i)
    result.push(namedObjects[i].getName());
  return result;
};

e = F.DG.createEvent("event");
console.log(e.getName());
console.log(e.getType());

console.log(JSON.stringify(mapNamedObjectsToNames(e.getEventHandlers())));
eh = F.DG.createEventHandler("eventHandler");
e.appendEventHandler(eh);
console.log(JSON.stringify(mapNamedObjectsToNames(e.getEventHandlers())));

console.log(e.getDidFireCallback());
e.setDidFireCallback(function () {
  console.log(this.getName() + " fired");
});
console.log(e.getDidFireCallback());
e.fire();

F.flush();

File: Native/Test/Python/event.py
import fabric
F = fabric.createClient()

def mapNamedObjectsToNames( objects ):
  result = []
  for i in range( 0, len( objects ) ):
    result.append( objects[i].getName() )
  return result

e = F.DG.createEvent("event")
print(e.getName())
print(e.getType())

print(fabric.stringify(mapNamedObjectsToNames(e.getEventHandlers())))
eh = F.DG.createEventHandler("eventHandler")
e.appendEventHandler(eh)
print(fabric.stringify(mapNamedObjectsToNames(e.getEventHandlers())))

print(e.getDidFireCallback())

def fireCallback( self ):
  print( self.getName() + " fired" )

e.setDidFireCallback( fireCallback )

print(type( e.getDidFireCallback() ))
e.fire()

F.close()
